src/m365/base/PowerPlatformCommand.ts

```diff
@@ -19,5 +20,7 @@ export default abstract class PowerPlatformCommand extends Command {
     if (auth.connection.cloudType !== CloudType.Public) {
       throw new CommandError(`Power Platform commands only support the public cloud at the moment. We'll add support for other clouds in the future. Sorry for the inconvenience.`);
     }
+
+    accessToken.ensureDelegatedAccessToken();
```

Comment:
Would like a second opinion if this is the right place to check for the delegated access token. I think it is but would like a second opinion from @waldekmastykarz.

Reply:
It's the right place. Given how the check works, I suggest we rename it to `assertDelegatedAccessToken`. _Ensure_ doesn't quite communicate what the method will do and you could think it might convert the token or get a new one, while it's not the case. _Assert_ on the other hand is clearer because if the check isn't valid, it fails with an exception, similarly to how asserts in tests do.

Reply:
That's what I was thinking as well (https://github.com/pnp/cli-microsoft365/pull/5862#discussion_r1600590000) 😊
Thanks for the review!